src/util/compat.js

```diff
@@ -43,3 +44,34 @@ export function fromCompat<T>(
   }
   return result;
 }
+
+const headerParser = C.object({type: C.string, version: C.string});
+const wrappedParser = C.tuple([headerParser, C.raw]);
+
+export function compatibleParser<T>(
+  expectedType: string,
+  handlers: {[version: string]: C.Parser<T>}
```

Comment:
Should be read-only: `{+[version: string]: C.Parser<T>}`.

Reply:
Done.

Reply:
Done.